google-cloud-core/src/main/java/com/google/cloud/ServiceOptions.java

```diff
@@ -388,6 +388,9 @@ private static boolean isWindows() {
 
   protected static String getAppEngineProjectId() {
     String projectId = null;
+    if (!PlatformInformation.isOnGAE()) {
```

Comment:
Can you check whether `PlatformInformation.isOnGAE()` works for GAE flex environ?